apps/persons/tests/utils.py

```diff
@@ -0,0 +1,124 @@
+        user_with_permission = UserFactory()
+        staff = UserFactory(is_staff=True)
+        user = UserFactory()
+        anonymous = AnonymousUser()
+
+        # Add global permission to change page for users concerned
+        can_change_page = Permission.objects.get(codename="change_page")
+        staff_with_permission.user_permissions.add(can_change_page)
+        user_with_permission.user_permissions.add(can_change_page)
+
+        cases = [
+            ([superuser, False, False], "disabled"),
+            ([superuser, True, False], "active"),
+            ([superuser, False, True], "disabled"),
+            ([staff_with_permission, False, False], "disabled"),
+            ([staff_with_permission, True, False], "active"),
+            ([staff_with_permission, False, True], "disabled"),
+            ([staff, False, False], "missing"),
+            ([staff, True, False], "missing"),
+            ([staff, False, True], "missing"),
+            ([user_with_permission, False, False], "absent"),
+            ([user_with_permission, True, False], "absent"),
+            ([user_with_permission, False, True], "absent"),
+            ([user, False, False], "absent"),
+            ([user, True, False], "absent"),
+            ([user, False, True], "absent"),
+            ([anonymous, False, False], "absent"),
+            ([anonymous, True, False], "absent"),
+            ([anonymous, False, True], "absent"),
+        ]
+        admin_url = "/en/admin/{app_name:s}/{model_name:s}/{id:d}/change/".format(
+            app_name=page_extension._meta.app_label,
+            model_name=page_extension.__class__.__name__.lower(),
+            id=page_extension.id,
+        )
+
+        page = page_extension.extended_object
+        for args, state in cases:
+            toolbar = get_toolbar_for_page(page, *args)
+
+            if state in ["active", "disabled", "missing"]:
+                page_menu_result = toolbar.find_items(Menu, name="Page")[0]
+                results = page_menu_result.item.find_items(
+                    ModalItem, name=menu_item_text
+                )
+                if state in ["active", "disabled"]:
+                    self.assertEqual(len(results), 1)
+                    item = results[0].item
+                    self.assertEqual(item.url, admin_url)
+
+                    if state == "active":
+                        self.assertFalse(item.disabled)
+
+                    elif state == "disabled":
+                        self.assertTrue(item.disabled)
+
+                elif state == "missing":
+                    self.assertEqual(results, [])
+
+                else:
+                    raise ImproperlyConfigured()
+
+            elif state == "absent":
+                # The whole toolbar should be hidden
+                self.assertEqual(toolbar.get_left_items(), [])
+
+            else:
+                raise ImproperlyConfigured()
+
+
+def get_toolbar_for_page(page, user, edit, preview):
```

Comment:
maybe make it a @staticmethod of your mixin ?

Reply:
It's can not be a static method as it refer to self to make some asserts just lower...
Therefor this factorization may be a bad design, WDYT ?

Reply:
no you're mixing up this method and the method above. This one does not need self.
I think this mixin is fine. It can have a normal method (check_toolbar_item) and a staticmethod (get_toolbar_for_page)